### ISSUE TITLE
when convert adj to lists

### ISSUE THREAD

**Issue by mmvv11** [BokwaiHo/DEGC#3]
Hi, thanks for ur study.

I’m just wondering if it should be `lists.append(list(i_list.nonzero()[1]))` rather than `lists.append(list(i_list.nonzero()[0]))` when converting adj to lists.

https://github.com/BokwaiHo/DEGC/blob/716a16d10103da17499ddc8642802f10c7b599c4/data_utils/utils.py#L155

**Comment by mmvv11** [comment]
since `lists.append(list(i_list.nonzero()[0]))` indicates user, i thought it should be `ists.append(list(i_list.nonzero()[1]))`

**Comment by BokwaiHo** [comment]
Hi Jiwoon:
  Thanks for your interest to our work. I have checked the code carefully and confirm that `lists.append(list(i_list.nonzero()[0]))` indicates the collection of items corresponding to the given user `u`. So, it should be `lists.append(list(i_list.nonzero()[0]))` as provided in the original code.
Thanks

**Comment by mmvv11** [comment]
thanks for ur answer!

But I still wondering that it should be approached as 1, not 0, because the rows mean user indexes and the columns mean item indexes.
Shouldn't the rows and columns of adj_matrix mean user and item respectively?

**Comment by BokwaiHo** [comment]
Thanks for your request. I confirm again that it should be `lists.append(list(i_list.nonzero()[0]))`. Yes, the rows mean user indexes and the columns mean item indexes. And the rows and columns of adj_matrix do mean user and item respectively. I suspect that you may not be familiar to the Numpy function `nonzero()` here. 
I can give your an example to help you understand how the code works here. 
`import numpy as np`
`i_list = np.array([1,1,0,1,0])`
` i_list
array([1, 1, 0, 1, 0])`
`i_list.nonzero()`
`(array([0, 1, 3]),)`
`i_list.nonzero()[0]`
`array([0, 1, 3])`
`i_list.nonzero()[1]`
`Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
IndexError: tuple index out of range`
Hope this code snippet can help understand this function better.  I don't know why you insist it should be approached as 1, not 0.  If you believe your are right, you can change the code according to your own idea and observe what will happen. Obviously, an error will occur. I have tried my best to clarify this point!!!

**Comment by mmvv11** [comment]
Thank you for your continued kind explanations.

However, I do understand the numpy behavior in question.
In fact, if I set it to 1 and run the code as I claimed, it works fine.
Although I agree with you about the example,
As I understand it, the adj_matrix parameter of the convert_adj_matrix_to_lists method is a csr matrix object.
Therefore, i claimed that it should be set to 1, not 0, so that the items can be sampled.

Anyway, I appreciate your efforts and respect it.
Have a nice day.
